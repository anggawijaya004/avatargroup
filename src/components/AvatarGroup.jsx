import React from "react";
import Avatar from "./Avatar";

export default function AvatarGroup({ users, maxLength, size }) {
  let valSize;
  switch (size) {
    case "xs":
      valSize = 16;
      break;
    case "sm":
      valSize = 24;
      break;
    case "md":
      valSize = 32;
      break;
    case "lg":
      valSize = 64;
      break;
    default:
      valSize = 32;
      break;
  }
  
  const style = {
    width: valSize,
    height: valSize,
    fontSize: `${valSize / 2}px`,
    marginLeft: `-${valSize / 4}px`,
  };

  const remain = users.length - maxLength;

  return (
    <div className="avatar-group">
      {users.slice(0, maxLength).map((name, i) => (
        <Avatar testId={`avatar-${i}`} avatarStyle={style} name={name} key={name} />
      ))}
      {remain > 0 && (
        <div className="avatar" style={style}>
          +{remain}
        </div>
      )}
    </div>
  );
}
