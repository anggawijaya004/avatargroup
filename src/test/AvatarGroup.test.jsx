import { it, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import AvatarGroup from "../components/AvatarGroup";

describe("Avatar Group", () => {
  const users = [
    "Calos Tevez",
    "Luis Suarez",
    "Edison Cavani",
    "Abdur Rasyid",
    "Najwa Syihab",
  ];
  const maxLength = 4;
  const size = "md";

  it("renders initials correctly", () => {
    render(<AvatarGroup users={users} maxLength={maxLength} size={size} />);
    expect(screen.getByTestId("avatar-0").textContent).toBe("CT");
    expect(screen.getByTestId("avatar-1").textContent).toBe("LS");
  });

  it("renders size correctly", () => {
    render(<AvatarGroup users={users} maxLength={maxLength} size={size} />);
    expect(getComputedStyle(screen.getByTestId("avatar-0")).width).toBe("32px");
  });

  it("renders size correctly", () => {
    render(<AvatarGroup users={users} maxLength={maxLength} size="md" />);
    expect(getComputedStyle(screen.getByTestId("avatar-0")).width).toBe("32px");
  });

  it("renders maxLength correctly", () => {
    render(<AvatarGroup users={users} maxLength={maxLength} size={size} />);
    expect(screen.getAllByRole("child").length).toBe(maxLength);
  });

  it("snapshot", () => {
   const result = render(<AvatarGroup users={users} maxLength={maxLength} size={size} />);
    expect(result).toMatchSnapshot();
  });
});
