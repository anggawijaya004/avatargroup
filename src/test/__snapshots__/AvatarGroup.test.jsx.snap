// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar Group > snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="avatar-group"
      >
        <div
          class="avatar"
          data-testid="avatar-0"
          role="child"
          style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
        >
          CT
        </div>
        <div
          class="avatar"
          data-testid="avatar-1"
          role="child"
          style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
        >
          LS
        </div>
        <div
          class="avatar"
          data-testid="avatar-2"
          role="child"
          style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
        >
          EC
        </div>
        <div
          class="avatar"
          data-testid="avatar-3"
          role="child"
          style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
        >
          AR
        </div>
        <div
          class="avatar"
          style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
        >
          +
          1
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="avatar-group"
    >
      <div
        class="avatar"
        data-testid="avatar-0"
        role="child"
        style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
      >
        CT
      </div>
      <div
        class="avatar"
        data-testid="avatar-1"
        role="child"
        style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
      >
        LS
      </div>
      <div
        class="avatar"
        data-testid="avatar-2"
        role="child"
        style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
      >
        EC
      </div>
      <div
        class="avatar"
        data-testid="avatar-3"
        role="child"
        style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
      >
        AR
      </div>
      <div
        class="avatar"
        style="width: 32px; height: 32px; font-size: 16px; margin-left: -8px;"
      >
        +
        1
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
